rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -2341,8 +2383,8 @@ rmw_destroy_service(rmw_node_t * node, rmw_service_t * service)
   // CLEANUP ================================================================
   z_drop(z_move(service_data->keyexpr));
   z_drop(z_move(service_data->qable));
-  for (auto & id_query : service_data->id_query_map) {
-    z_drop(z_move(id_query.second));
+  for (z_owned_query_t & query : service_data->query_queue) {
```

Comment:
@clalancette since we move ownership of taken requests from `query_queue` to `sequence_id_map`, should we also iterate over the map and properly drop the `z_owned_query_t`?